.github/workflows/java.yml

```diff
@@ -0,0 +1,27 @@
+name: Java
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+
+jobs:
+  rewriteCatch:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup C++
+        uses: ./.github/actions/setup-cpp
+
+      - name: Setup Java
+        uses: ./.github/actions/setup-java
+
+      - name: Build slice2java
+        run: make -C cpp slice2java
+
+      - name: Run rewriteDryRun with build failure enabled
+        run: ./java/gradlew rewriteDryRun -PenableDryRunFail -PenableCops --project-dir ./java/
```

Comment:
I would set `working-directory: java` and simplify the command to `./gradlew rewriteDryRun -PenableDryRunFail`